usaspending_api/agency/v2/views/budget_function.py

```diff
@@ -0,0 +1,89 @@
+from django.db.models import Q, Sum
+from rest_framework.request import Request
+from rest_framework.response import Response
+from typing import Any
+from usaspending_api.agency.v2.views.agency_base import AgencyBase, ListMixin
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.financial_activities.models import FinancialAccountsByProgramActivityObjectClass
+
+
+class BudgetFunctionList(ListMixin, AgencyBase):
+    """
+    Obtain the count of budget functions for a specific agency in a single
+    fiscal year based on whether or not that budget function has ever
+    been submitted in File B.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/budget_function.md"
+
+    def format_results(self, rows):
+        order = self.pagination.sort_order == "desc"
+        names = set([row["treasury_account__budget_function_title"] for row in rows])
+        budget_functions = [{"name": x} for x in names]
+        for item in budget_functions:
+            item["children"] = [
+                {
+                    "name": row["treasury_account__budget_subfunction_title"],
+                    "obligated_amount": row["obligated_amount"],
+                    "gross_outlay_amount": row["gross_outlay_amount"],
+                }
+                for row in rows
+                if item["name"] == row["treasury_account__budget_function_title"]
+            ]
+            item["obligated_amount"] = sum([x["obligated_amount"] for x in item["children"]])
+            item["gross_outlay_amount"] = sum([x["gross_outlay_amount"] for x in item["children"]])
+            item["children"] = sorted(item["children"], key=lambda x: x[self.pagination.sort_key], reverse=order)
+        budget_functions = sorted(budget_functions, key=lambda x: x[self.pagination.sort_key], reverse=order)
+        return budget_functions
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        results = self.format_results(list(self.get_budget_function_queryset()))
+        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        return Response(
+            {
+                "toptier_code": self.toptier_code,
+                "fiscal_year": self.fiscal_year,
+                "results": (results[self.pagination.lower_limit : self.pagination.upper_limit])[
+                    : self.pagination.limit
+                ],
```

Comment:
```suggestion
                "results": results[self.pagination.lower_limit : self.pagination.upper_limit]],
```

The +1 was removed from the upper_limit attribute in another PR

Reply:
oh super

Reply:
Opps, I left a trailing square bracket....